sources/packages/backend/libs/integrations/src/institution-integration/ece-integration/ece-response.processing.service.ts

```diff
@@ -312,6 +332,51 @@ export class ECEResponseProcessingService {
     return disbursementProcessingDetails;
   }
 
+  /**
+   * Validate disbursement data.
+   * @param disbursement disbursement.
+   */
+  private validateDisbursement(disbursement: DisbursementDetails): void {
+    const errors: string[] = [];
+
+    const hasInvalidEnrolmentConfirmationFlag = disbursement.awardDetails.some(
+      (award) => !["Y", "N"].includes(award.enrolmentConfirmationFlag),
+    );
+
+    if (hasInvalidEnrolmentConfirmationFlag) {
+      errors.push("Invalid enrolment confirmation flag");
+    }
+
+    const confirmedEnrolments = disbursement.awardDetails.filter(
+      (award) => award.enrolmentConfirmationFlag === "Y",
+    );
+
+    // Validate the enrolment confirmation date for confirmed enrolments.
+    const hasInvalidEnrolmentConfirmationDate = confirmedEnrolments.some(
+      (award) => award.enrolmentConfirmationDate.toString() === "Invalid Date",
```

Comment:
I see that `Invalid Date` here is the result of the `dayjs` attempt to convert a date. Maybe we can have it as a const in the `date-utils.is` or have a small method `isValidDate` to check for this.

Reply:
The `Invalid Date` is from the native JS/TS date object.

![image](https://github.com/bcgov/SIMS/assets/54600590/90fda494-03ea-4c07-8a6a-d56e646313e8)

I can create a small function in date utils to check valid date.

Reply:
Just noticed that dayjs has a method isValid which we can use. I can create an util with that.